artifacts/definitions/Windows/EventLogs/PowershellScriptblock.yaml

```diff
@@ -51,58 +57,86 @@ parameters:
+  - query: |
+        -- Build time bounds
+        LET DateAfterTime <= if(condition=DateAfter, 
+            then=timestamp(epoch=DateAfter), else=timestamp(epoch="1600-01-01"))
+        LET DateBeforeTime <= if(condition=DateBefore, 
+            then=timestamp(epoch=DateBefore), else=timestamp(epoch="2200-01-01"))
+
+        -- Parse Log level dropdown selection
+        LET LogLevelRegex <= SELECT format(format="%v", args=Regex) as value
             FROM parse_csv(filename=LogLevelMap, accessor="data")
             WHERE Choice=LogLevel LIMIT 1
-      - LET files = SELECT * FROM glob(
-            globs=eventLog)
-      - SELECT *
-        FROM foreach(
-          row=files,
-          query={
-            SELECT timestamp(epoch=System.TimeCreated.SystemTime) As EventTime,
-              System.Computer as Computer,
-              System.Security.UserID as SecurityID,
-              EventData.Path as Path,
-              EventData.ScriptBlockId as ScriptBlockId,
-              EventData.ScriptBlockText as ScriptBlockText,
-              System.EventRecordID as EventRecordID,
-              System.Level as Level,
-              System.Opcode as Opcode,
-              System.Task as Task
-            FROM parse_evtx(filename=FullPath)
-            WHERE System.EventID.Value = 4104 and
-                if(condition=dateAfter, then=EventTime > timestamp(string=dateAfter),
-                 else=TRUE) and
-                if(condition=dateBefore, then=EventTime < timestamp(string=dateBefore),
-                 else=TRUE) and
-                format(format="%d", args=System.Level) =~ LogLevelRegex.value[0] and
-                if(condition=searchStrings, then=ScriptBlockText =~ searchStrings,
-                 else=TRUE) and
-                if(condition=stringWhitelist, then=not ScriptBlockText =~ stringWhitelist,
-                 else=TRUE) and
-                if(condition=pathWhitelist, then=not Path =~ pathWhitelist,
-                 else=TRUE)
-        })
 
-reports:
-  - type: HUNT
-    template: |
-      Powershell: Scriptblock
-      =======================
-      Powershell is commonly used by attackers accross all stages of the attack 
-      lifecycle.  
-      A valuable hunt is to search Scriptblock logs for signs of malicious 
-      content. Stack ranking these events can provide valuable leads from which 
-      to start an investigation.
-      
-      {{ Query "SELECT count(items=ScriptBlockText) as Count, ScriptBlockText FROM source(source='PowershellScriptBlock') GROUP BY ScriptBlockText ORDER BY Count"  | Table }}
-      
\ No newline at end of file
+        -- Determine target files
+        LET files = SELECT *
+          FROM if(condition=SearchVSS,
+            then= {
+              SELECT *
+              FROM Artifact.Windows.Search.VSS(SearchFilesGlob=EventLog)
+            },
+            else= {
+              SELECT *
+              FROM glob(globs=EventLog,accessor='ntfs')
```

Comment:
Its probably not super necessary to use ntfs here - file is generally faster.

Reply:
just bounced this in